test/e2e/test_cli.py

```diff
@@ -205,15 +205,17 @@ def test_xml_upload_incremental(self) -> None:
             working_directory=Path("."),
         )
 
-        mapping_file = list(Path().glob("test-data-systematic_id2iri_mapping_*.json"))[0]
+        mapping_file_candidates = Path().glob("test-data-systematic_id2iri_mapping_*.json")
+        mapping_file = sorted(mapping_file_candidates)[-1]  # most recent one
         second_xml_file_orig = Path("testdata/id2iri/test-id2iri-data.xml")
         self._make_cli_call(f"dsp-tools id2iri {second_xml_file_orig.absolute()} {mapping_file.absolute()}")
+        mapping_file.unlink()
 
-        second_xml_file_replaced = list(self.cwd.glob(f"{second_xml_file_orig.stem}_replaced_*.xml"))[0]
+        second_xml_file_replaced_candidates = self.cwd.glob(f"{second_xml_file_orig.stem}_replaced_*.xml")
+        second_xml_file_replaced = sorted(second_xml_file_replaced_candidates)[-1]  # most recent one
```

Comment:
You are doing the same thing twice 208-212 and 214 and again in 268, maybe you could make a function for this?

Reply:
better now?